src/js/dynload.ts

```diff
@@ -97,6 +114,19 @@ export async function loadDynlib(lib: string, global: boolean) {
       global,
       fs,
     });
+
+    // Emscripten saves the list of loaded libraries in LDSO.loadedLibsByName.
+    // However, since emscripten dylink metadata only contains the name of the
+    // library not the full path, we need to update it manually in order to
+    // prevent loading same library twice.
+    if (global && Module.PATH.isAbs(lib)) {
+      const libName: string = Module.PATH.basename(lib);
+      const dso: any = Module.LDSO.loadedLibsByName[libName];
+      if (!dso) {
+        Module.LDSO.loadedLibsByName[libName] =
```

Comment:
Funny we used to have to do the same thing with `preloadedWasm`, then `preloadedWasm` got removed and now we have to add the same hack with the new `loadedLibsByName`.

Reply:
Yeah, while writing this PR, I noticed that `preloadedWasm` hack was doing the same thing... my bad.

Reply:
Well this is an improvement in a lot of other ways over using `preloadedWasm`, and at least you tried removing the hack and learned that it wouldn't go away.